web/src/user_status.ts

```diff
@@ -21,16 +21,35 @@ const user_status_event_schema = z.intersection(
     user_status_schema,
 );
 
+export type TimeKey =
+    | "never"
+    | "in_thirty_minutes"
+    | "in_one_hour"
+    | "today_five_pm"
+    | "tomorrow"
+    | "custom";
+
 export type UserStatusEvent = z.infer<typeof user_status_event_schema>;
 
 const user_info = new Map<number, string>();
 const user_status_emoji_info = new Map<number, UserStatusEmojiInfo>();
 
+let scheduled_end_time: number | null = null;
+
+export function set_scheduled_end_time(scheduled_time: number | null): void {
+    scheduled_end_time = scheduled_time;
+}
+
+export function get_scheduled_end_time(): number | null {
+    return scheduled_end_time;
+}
+
 export function server_update_status(opts: {
     status_text: string;
     emoji_name: string;
     emoji_code: string;
     reaction_type?: string;
+    scheduled_end_time: number | string;
```

Comment:
This should take `number | null`, and should take responsibility for `JSON.stringify`ing it here, not in the caller.